pkg/reconciler/route/route.go

```diff
@@ -318,16 +320,32 @@ func (c *Reconciler) tls(ctx context.Context, host string, r *v1alpha1.Route, tr
 		}
 	}
 
+	routeDomain := config.FromContext(ctx).Domain.LookupDomainForLabels(r.Labels)
+	labelSelector := kubelabels.SelectorFromSet(
+		kubelabels.Set{
+			networking.WildcardCertDomainLabelKey: routeDomain,
+		},
+	)
+
+	allCerts, err := c.certificateLister.Certificates(r.Namespace).List(labelSelector)
+	if err != nil {
+		return nil, err
+	}
+
 	desiredCerts := resources.MakeCertificates(r, tagToDomainMap, certClass(ctx, r))
 	for _, desiredCert := range desiredCerts {
+		dnsNames := sets.NewString(desiredCert.Spec.DNSNames...)
+		cert := findMatchingCert(desiredCert.Spec.DNSNames, allCerts)
```

Comment:
`wildcardCert := ...`

Reply:
I guess I could see why you'd leave this as is, but perhaps a comment indicating why we'd want to bias towards a wildcard by default would be useful.

Reply:
I'll add a comment for sure but if there is a better way to structure this logic I can change it.